toolkit/analysis_and_visualization/classification/generate-visualization.py

```diff
@@ -59,2520 +79,887 @@ def readHierarchyFile(hierarchyFile):
-            if((prob_4 <= (f + 0.01)) and prob_4 > f):
-              top5PassFail[count][7]+=1;
-              hierarchyPenalty[count][3] += calculateHierarchyPenalty(truth,label_4,hierarchyDataBase);
-            if((prob_5 <= (f + 0.01)) and prob_5 > f):
-              top5PassFail[count][8]+=1;
-            count+=1;
-            f+=0.01;
+            previousTruth = 1
+        elif((previousTruth == 1) and (token_truth == '' and token_result == '')):
+            previousTruth = 1
         else:
-          count = 0;
-          f = 0;
-          while f < 1:
-            if((prob_1 <= (f + 0.01)) and prob_1 > f):
-              top5PassFail[count][1]+=1;
-              hierarchyPenalty[count][0] += calculateHierarchyPenalty(truth,label_1,hierarchyDataBase);
-            if((prob_2 <= (f + 0.01)) and prob_2 > f):
-              top5PassFail[count][3]+=1;
-              hierarchyPenalty[count][1] += calculateHierarchyPenalty(truth,label_2,hierarchyDataBase);
-            if((prob_3 <= (f + 0.01)) and prob_3 > f):
-              top5PassFail[count][5]+=1;
-              hierarchyPenalty[count][2] += calculateHierarchyPenalty(truth,label_3,hierarchyDataBase);
-            if((prob_4 <= (f + 0.01)) and prob_4 > f):
-              top5PassFail[count][7]+=1;
-              hierarchyPenalty[count][3] += calculateHierarchyPenalty(truth,label_4,hierarchyDataBase);
-            if((prob_5 <= (f + 0.01)) and prob_5 > f):
-              top5PassFail[count][9]+=1;
-              hierarchyPenalty[count][4] += calculateHierarchyPenalty(truth,label_5,hierarchyDataBase);
-            count+=1;
-            f+=0.01;
-
-    print("<A NAME=\"table6\"><h1 align=\"center\"><font color=\"DodgerBlue\" size=\"6\"><br><br><br><em>Model Score</em></font></h1></A>");
-    print("\t");
-    Top1PassScore = 0;
-    Top1FailScore = 0;
-    Top2PassScore = 0;
-    Top2FailScore = 0;
-    Top3PassScore = 0;
-    Top3FailScore = 0;
-    Top4PassScore = 0;
-    Top4FailScore = 0;
-    Top5PassScore = 0;
-    Top5FailScore = 0;
-    Top1HierarchyPenalty = 0;
-    Top2HierarchyPenalty = 0;
-    Top3HierarchyPenalty = 0;
-    Top4HierarchyPenalty = 0;
-    Top5HierarchyPenalty = 0;
-    confID=0.99;
-    i = 99;
+            previousTruth = 0
+            penaltyMultiplier += 1
+        catCount += 1
+
+    penaltyMultiplier = float(penaltyMultiplier - 1)
+    penaltyValue = (0.2 * penaltyMultiplier)
+
+    return penaltyValue
+
+
+def createScoreSummary(stats, topCounts):
+    logger.debug('Calculating scores')
+
+    topk = 5
+    topScores = [float(t) for t in topCounts]
+    modelScores = [0.0] * topk
+    matchCounts = [0] * topk
+
+    for i in range(topk):
+        modelScores[i] = (sum(topScores[:i+1]) /
+                          stats['netSummaryImages']) * 100.0
+        matchCounts[i] = sum(topCounts[:i+1])
+
+    return modelScores, matchCounts
+
+
+def createHirerchySummaryScore(stats, topCounts, resultDataBase, labelLines, hierarchyDataBase, topKPassFail):
```

Comment:
Codacy found an issue: [createHirerchySummaryScore is too complex (41) (MC0001)](https://app.codacy.com/gh/GPUOpen-ProfessionalCompute-Libraries/MIVisionX/pullRequest?prid=6397592)